packages/bitcore-node/src/modules/ethereum/api/csp.ts

```diff
@@ -176,19 +189,13 @@ export class ETHStateProvider extends InternalStateProvider implements IChainSta
       let query = { chain, network, txid: txId };
       const tip = await this.getLocalTip(params);
       const tipHeight = tip ? tip.height : 0;
-      const found = await EthTransactionStorage.collection.findOne(query);
+      let found = await EthTransactionStorage.collection.findOne(query);
       if (found) {
         let confirmations = 0;
         if (found.blockHeight && found.blockHeight >= 0) {
           confirmations = tipHeight - found.blockHeight + 1;
         }
-        if (!found.receipt) {
-          const receipt = await this.getReceipt(network, found.txid);
-          if (receipt) {
-            await EthTransactionStorage.collection.updateOne({ _id: found._id }, { $set: { receipt } });
-            found.receipt = receipt;
-          }
-        }
+        found = await this.populateReceipt(found);
```

Comment:
Do we need to return the `tx` here? Wouldn't be more clear to do just do:
```
   await this.populateReceipt(found);
```

Reply:
That would work as well. I do this to signal that found is being changed.

Reply:
Great work this is going to make everyone happy. If you are working diligently on any more projects I am doing please speak up about sponsoring. I don't know how that originates but I need as many of you on my team as possible. Kaycoins@outlook.com.